File: leetcode/java/0141. Linked List Cycle.java
package leetcode.java;

// Time complexity: O(N)
// Space complexity: O(1)

//Provided definition
class ListNode {
    int val;
    ListNode next;
    ListNode(int x) {
        val = x;
        next = null;
    }
}


class Solution {
    public boolean hasCycle(ListNode head) {
        while (head != null) {
            if (head.val == Integer.MAX_VALUE) {
                return true;
            } else {
                head.val = Integer.MAX_VALUE;
            }
            head = head.next;
        }
        return false;
    }
}

File: leetcode/python/0141. Linked List Cycle.py
# Time complexity: O(N)
# Space complexity: O(1)

class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution(object):
    def hasCycle(self, head):
        while head != None:
            if head.val is True:
                return True
            head.val = True
            head = head.next
        return False